### PR TITLE
Adding KubeMQ Binding - Queue

### PR DESCRIPTION
**Pull request by kubemq**
Signed-off-by: Lior Nabat <lior.nabat@gmail.clom>

# Description
This PR is to add KubeMQ Binding - Queue feature to DAPR.
This PR passed:
1. make lint
2. make test
3. go test -v -tags=conftests -count=1 ./tests/conformance -run="TestBindingsConformance/kubemq"

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ x] Code compiles correctly
* [ x] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by yaron2** [comment]
Almost there! @kubemq see the linter errors, some files show `File is not `gofumpt`-ed (gofumpt)`.

**Comment by kubemq** [comment]
@yaron2,
These files are not mine,
They are from master branch

**Comment by berndverst** [comment]
> @yaron2, These files are not mine, They are from master branch

The linter error is definitely happening on your file :)

File: bindings/kubemq/kubemq.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	if err != nil {
+		return nil, err
+	}
+	if len(result.Results) > 0 {
+		if result.Results[0].IsError {
+			return nil, fmt.Errorf("error sending queue message: %s", result.Results[0].Error)
+		}
+	}
+	return &bindings.InvokeResponse{
+		Data:        nil,
```
Comment (DeepanshuA) [inline bindings/kubemq/kubemq.go:94]: Just wanted to cross-check that nothing needs to be set in `Data` from `result`.

**Review by berndverst — CHANGES_REQUESTED** [review]
Can you also add it to the `.github/workflow/conformance.yaml` workflow file please? Just like you did for the pubsub component. Thanks!

**Comment by kubemq** [comment]
very confused.
when i do make lint i get:

bindings/aws/s3/s3.go:1                         gofmt    File is not `gofmt`-ed with `-s`
bindings/aws/s3/s3_test.go:1                    gofmt    File is not `gofmt`-ed with `-s`
bindings/aws/s3/s3_test.go:168                  gofumpt  File is not `gofumpt`-ed
bindings/zeebe/jobworker/jobworker.go:1         gofmt    File is not `gofmt`-ed with `-s`
bindings/zeebe/jobworker/jobworker_test.go:113  gofumpt  File is not `gofumpt`-ed
internal/utils/utils.go:49                      gofumpt  File is not `gofumpt`-ed

not related to me.
For sure i miss something.

File: bindings/kubemq/kubemq.go (hunk 1 above)
Comment (kubemq) [reply bindings/kubemq/kubemq.go:94]: This is sending message to queue,
nothing is returned unless error

File: bindings/kubemq/kubemq.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+		Metadata:    nil,
+		ContentType: nil,
```
[the remark this answers is not in the document]
Comment (kubemq) [reply bindings/kubemq/kubemq.go:96]: removed,

**Comment by DeepanshuA** [comment]
> very confused. when i do make lint i get:
> 
> bindings/aws/s3/s3.go:1 gofmt File is not `gofmt`-ed with `-s` bindings/aws/s3/s3_test.go:1 gofmt File is not `gofmt`-ed with `-s` bindings/aws/s3/s3_test.go:168 gofumpt File is not `gofumpt`-ed bindings/zeebe/jobworker/jobworker.go:1 gofmt File is not `gofmt`-ed with `-s` bindings/zeebe/jobworker/jobworker_test.go:113 gofumpt File is not `gofumpt`-ed internal/utils/utils.go:49 gofumpt File is not `gofumpt`-ed
> 
> not related to me. For sure i miss something.

In case, it helps: My golangci-lint has version 1.48.0 and when I run `golangci-lint run`, it comes clean.

**Comment by berndverst** [comment]
> very confused. when i do make lint i get:
> 
> bindings/aws/s3/s3.go:1 gofmt File is not `gofmt`-ed with `-s` bindings/aws/s3/s3_test.go:1 gofmt File is not `gofmt`-ed with `-s` bindings/aws/s3/s3_test.go:168 gofumpt File is not `gofumpt`-ed bindings/zeebe/jobworker/jobworker.go:1 gofmt File is not `gofmt`-ed with `-s` bindings/zeebe/jobworker/jobworker_test.go:113 gofumpt File is not `gofumpt`-ed internal/utils/utils.go:49 gofumpt File is not `gofumpt`-ed
> 
> not related to me. For sure i miss something.

Hmm, I'm not sure why you are seeing that.

Keep an eye on the `Build linux_amd64 binaries` task which actually performs the linter check. I bet the failures won't appear there.

https://github.com/dapr/components-contrib/actions/runs/3447595663/jobs/5753765821

**Comment by kubemq** [comment]
> https://github.com/dapr/components-contrib/actions/runs/3447595663/jobs/5753765821

Looked on the raw data,
2022-11-11T19:49:55.1739144Z ##[group]run golangci-lint
2022-11-11T19:49:55.1739922Z Running [/home/runner/golangci-lint-1.48.0-linux-amd64/golangci-lint run --out-format=github-actions --timeout 15m] in [] ...
2022-11-11T19:52:44.0468807Z ##[error]File is not `gofumpt`-ed (gofumpt)
2022-11-11T19:52:44.0481336Z ##[error]File is not `gofumpt`-ed (gofumpt)
2022-11-11T19:52:44.0483727Z ##[error]File is not `gofumpt`-ed (gofumpt)
2022-11-11T19:52:44.0484358Z 

No indication which file....

**Comment by berndverst** [comment]
```
bindings/kubemq/kubemq.go:37    gofumpt  File is not `gofumpt`-ed
bindings/kubemq/options.go:118  gofumpt  File is not `gofumpt`-ed
bindings/kubemq/options.go:134  gofumpt  File is not `gofumpt`-ed
```

**Comment by kubemq** [comment]
> ```
> bindings/kubemq/kubemq.go:37    gofumpt  File is not `gofumpt`-ed
> bindings/kubemq/options.go:118  gofumpt  File is not `gofumpt`-ed
> bindings/kubemq/options.go:134  gofumpt  File is not `gofumpt`-ed
> ```

How i can see that ?
no matter what i do, i can't get rid of these errors.

**Comment by ItalyPaleAle** [comment]
@kubemq have you tried using [`gofumpt`](https://github.com/mvdan/gofumpt)? note that's different from `gofmt`

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
Please open a docs PR against the v1.10 branch to add this component and document the metadata options.

**Pull request merged**